### ISSUE TITLE
Save and load hdf5 representation

### ISSUE THREAD

**Issue by niekdejonge** [matchms/matchms#632]
By saving and loading spectra in HDF5 it allows you to only access a few spectra at any position in a list of spectra. Storing spectra in this format is especially helpful for training models, since there is no need to have all spectra in memory. 

It stores the spectra like same size vectors. This means the spectra need to have a predefined number of fragments and if they have less, they need to be padded with zeroes. But due to efficient compression this is not really a memory issue. 

@roman-bushuiev implemented something similar already for https://github.com/roman-bushuiev/DreaMS and will try to implement it for matchms, so make the DreaMS package build on matchms.

**Comment by acquayefrank** [comment]
@niekdejonge  is this still relevant? I  could work on this in the coming days.

**Comment by niekdejonge** [comment]
Yes certainly! @roman-bushuiev implemented something like this in https://github.com/pluskal-lab/DreaMS 
@roman-bushuiev were you already planning to work on this or could you otherwise maybe point @acquayefrank to the relevant code?

**Comment by roman-bushuiev** [comment]
Hi! @acquayefrank , it would be perfect if you could simplify [the MSData class](https://github.com/pluskal-lab/DreaMS/blob/ef53d44e027aaed2008bfafa0e85ce809de36409/dreams/utils/data.py#L177) from the DreaMS codebase and integrate it into matchms. The class converts standard data formats to .hdf5 and provides various methods to work with the data from .hdf5.
